src/Unifier.ml

```diff
@@ -160,6 +205,14 @@ exception Unify of variable * variable
 exception UnifySkolemInternal
 exception UnifySkolem of variable * variable
 
+(* FIXME FEMRICH: We probably want to add more info to this, to aid error
+  messages.
+  It can't be just a term variable: We may unify two type vars that are both
+  monomorphic and whose mono constraints originate from different term
+  variables. Further, if we consider non-generalizing let at some point, there
+  isn't a term variable imposing the monomorphism. *)
+exception UnifyMono
```

Comment:
Not sure what you mean by "it happens to be the scheme". I don't see how a type scheme is available when we raise `UnifyMono`.

Reply:
Maybe a bit imprecise on my side, but:

```
  match s_opt with
  | None   -> ()
  | Some s ->
     if S.isForall s then
       raise UnifyMono
```

I was thinking about the `s`. I just realized that this is actually a structure, not a variable, so now I'm not sure if it can be placed (easily) in an error message.

Reply:
let's eschew it for now then